janusgraph-core/src/main/java/org/janusgraph/core/attribute/GeoshapeSerializer.java

```diff
@@ -0,0 +1,203 @@
+            int len= Array.getLength(value);
+            double[] arr = new double[len];
+            for (int i=0;i<len;i++) arr[i]=((Number)Array.get(value,i)).doubleValue();
+            switch (len) {
+                case 2:
+                    return Geoshape.point(arr[0], arr[1]);
+                case 3:
+                    return Geoshape.circle(arr[0], arr[1], arr[2]);
+                case 4:
+                    return Geoshape.box(arr[0], arr[1], arr[2], arr[3]);
+                default:
+                    throw new IllegalArgumentException("Expected 2-4 coordinates to create Geoshape, but given: " + value);
+            }
+        } else if (value instanceof String) {
+            String[] components=null;
+            for (String delimiter : new String[]{",",";"}) {
+                components = ((String)value).split(delimiter);
+                if (components.length>=2 && components.length<=4) break;
+                else components=null;
+            }
+            Preconditions.checkNotNull(components, "Could not parse coordinates from string: %s",value);
+            double[] coordinates = new double[components.length];
+            try {
+                for (int i=0;i<components.length;i++) {
+                    coordinates[i]=Double.parseDouble(components[i]);
+                }
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Could not parse coordinates from string: " + value, e);
+            }
+            return convert(coordinates);
+        } else return null;
+    }
+
+
+    private double[] convertCollection(Collection<Object> c) {
+
+        List<Double> numbers = c.stream().map(o -> {
+            if (!(o instanceof Number)) {
+                throw new IllegalArgumentException("Collections may only contain numbers to create a Geoshape");
+            }
+            return ((Number) o).doubleValue();
+        }).collect(Collectors.toList());
+        return Doubles.toArray(numbers);
+    }
+
+    private Geoshape convertGeoJson(Object value) {
+        //Note that geoJson is long,lat
+        try {
+            Map<String, Object> map = (Map) value;
+            String type = (String) map.get("type");
+            if("Feature".equals(type)) {
+                Map<String, Object> geometry = (Map) map.get("geometry");
+                return convertGeometry(geometry);
+            } else {
+                return convertGeometry(map);
+            }
+        } catch (ClassCastException | IOException | ParseException e) {
+            throw new IllegalArgumentException("GeoJSON was unparsable");
+        }
+    }
+
+    private Geoshape convertGeometry(Map<String, Object> geometry) throws IOException, ParseException {
+        String type = (String) geometry.get("type");
+        List<Object> coordinates = (List) geometry.get(Geoshape.FIELD_COORDINATES);
+
+        switch (type) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=3377550)